helm/jsonapi/templates/stargate_jsonapi_deployment.yaml

```diff
@@ -1,62 +1,195 @@
-      initContainers:
-      - name: wait-for-bridge
-        image: busybox
-        command: ["sh", "-c", "until nc -z {{ .Values.sgGrpcBridge.serviceName }}.{{ .Values.sgGrpcBridge.namespace }} 8091 > /dev/null; do echo Waiting for bridge.; sleep 5;done;"]
+            - topologyKey: {{ .Values.topologyKey }}
+              labelSelector:
+                matchLabels:
+                  app.kubernetes.io/component: jsonapi
       containers:
-      - name: jsonapi
-        image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}"
-        ports:
-        - containerPort: 8181
-        readinessProbe:
-          httpGet:
-            path: {{ .Values.probe.readiness.url }}
-            port: {{ .Values.probe.readiness.port }}
-          timeoutSeconds: {{ .Values.probe.readiness.timeoutSeconds }}
-          periodSeconds: {{ .Values.probe.readiness.periodSeconds }}
-          failureThreshold: {{ .Values.probe.readiness.failureThreshold }}
-          initialDelaySeconds: {{ .Values.probe.readiness.initialDelaySeconds }}
-        livenessProbe:
-          httpGet:
-            path: {{ .Values.probe.liveness.url }}
-            port: {{ .Values.probe.liveness.port }}
-          timeoutSeconds: {{ .Values.probe.liveness.timeoutSeconds }}
-          periodSeconds: {{ .Values.probe.liveness.periodSeconds }}
-          failureThreshold: {{ .Values.probe.liveness.failureThreshold }}
-          initialDelaySeconds: {{ .Values.probe.liveness.initialDelaySeconds }}
-        resources:
-          requests:
-            cpu: {{ .Values.cpu | default 1000 }}m
-            memory: {{ .Values.memory | default 1024 }}Mi
-        env:
-        - name: QUARKUS_GRPC_CLIENTS_BRIDGE_HOST
-          value: "{{ .Values.sgGrpcBridge.serviceName }}.{{ .Values.sgGrpcBridge.namespace }}"
-        - name: IO_STARGATE_SGV2_API_COMMON_PROPERTIES_DATASTORE_CONFIGURATION_DATASTOREPROPERTIESCONFIGURATION_FETCHSUPPORTEDFEATURES_RETRY_MAXRETRIES
-          value: "50"
-        - name: IO_STARGATE_SGV2_API_COMMON_PROPERTIES_DATASTORE_CONFIGURATION_DATASTOREPROPERTIESCONFIGURATION_FETCHSUPPORTEDFEATURES_RETRY_MAXDURATION
-          value: "300"
-        - name: IO_STARGATE_SGV2_API_COMMON_PROPERTIES_DATASTORE_CONFIGURATION_DATASTOREPROPERTIESCONFIGURATION_FETCHSUPPORTEDFEATURES_RETRY_DELAY
-          value: "10"
+        - env:
+            - name: QUARKUS_HTTP_HOST
+              value: "0.0.0.0"
+            - name: QUARKUS_HTTP_PORT
+              value: "{{ .Values.api.port }}"
+            - name: QUARKUS_SHUTDOWN_TIMEOUT
+              value: "PT30S"
+            - name: QUARKUS_SMALLRYE_OPENAPI_SERVERS
+              value: /api/json
+            - name: QUARKUS_HTTP_ACCESS_LOG_ENABLED
+              value: "true"
+            - name: QUARKUS_HTTP_ACCESS_LOG_PATTERN
+              value: "{{ .Values.accessLogPattern }}"
+            - name: QUARKUS_LOG_MIN_LEVEL
+              value: "TRACE"
```

Comment:
Do we really want to set this to "TRACE"? That level is potentially very verbose isn't it? Default is, I think, "DEBUG".

Reply:
Setting MIN_LEVEL to TRACE really helps when we want to find something with trace enabled. Without this, when required, even though we set the level to TRACE for a package or class, it will only be go up to DEBUG level.

Reply:
Ok so there's some pre-filtering done or something? I wasn't 100% sure what 2 separate levels meant.

Reply:
Yes, there are two separate levels. But I just noticed that `QUARKUS_LOG_MIN_LEVEL` is a build time property and so we will have to put in the application.yaml and passing it here as an environment generates a warning message on the start up.

https://quarkus.io/guides/all-config#quarkus-core_quarkus-log-min-level

Reply:
I think that might explain logic:

1. Min level determines log statements that are actually compiled in (so can statically remove unused trace level statements, if so desired)
2. Actual level used will dynamically determine if log statement is executed at run time

so just include TRACE in minimum does not enable trace-level output but allows if configure.

So basically what you said @kathirsvn  :)

Reply:
Thanks, I have set log_min_level in configuration file, please check